src/files/XmlFile.ts

```diff
@@ -458,27 +463,54 @@ export class XmlFile {
         this.program.logger.debug('XmlFile', chalk.green(this.pkgPath), ...args);
     }
 
+    private determinePkgMapPath(uri) {
+        if (/^pkg:\/.+/.test(uri)) {
+            return uri.replace(/^pkg:\//, '');
+        }
+        return util.getPkgPathFromTarget(this.pkgPath, uri);
+    }
```

Comment:
I think you can completely eliminate this function and just call `util.getPkgPathFromTarget` from `checkScriptsForPublishableImports`. We do a similar `pkg:/` check in `getPkgPathFromTarget`.